ckanext/canada/tables/aistrategy.yaml

```diff
@@ -0,0 +1,818 @@
+    excel_required: false
+    excel_column_width: 80
+    format_type: Free text
+    validation:
+      en: Limit the number of characters, free text (maximum 1000 characters).
+      fr: Limiter le nombre de caractères, libre texte (maximum 1000 caractères).
+    occurrence: Single
+    datastore_type: text
+    max_chars: 1000
+    form_snippet: scheming/form_snippets/textarea.html
+    form_attrs:
+      maxlength: 1000
+
+  # DEP 1-13
+  - datastore_id: progress_fr
+    label:
+      en: Progress Made (French)
+      fr: Progrès réalisés (français)
+    description:
+      en: Describes any progress made
+      fr: Décrit les progrès réalisés
+    obligation: Optional
+    excel_required: false
+    excel_column_width: 80
+    format_type: Free text
+    validation:
+      en: Limit the number of characters, free text (maximum 1000 characters).
+      fr: Limiter le nombre de caractères, libre texte (maximum 1000 caractères).
+    occurrence: Single
+    datastore_type: text
+    max_chars: 1000
+    form_snippet: scheming/form_snippets/textarea.html
+    form_attrs:
+      maxlength: 1000
+
+  # Dev: req dynamic field
+  - datastore_id: record_created
+    label: Record Creation Time
+    import_template_include: false
+    visible_to_public: false
+    datastore_type: timestamp
+    preview_class: bg-info
+
+  # Dev: req dynamic field
+  - datastore_id: record_modified
+    label: Last Record Modification Time
+    import_template_include: false
+    visible_to_public: false
+    datastore_type: timestamp
+    preview_class: bg-info
+
+  # Dev: req dynamic field
+  - datastore_id: user_modified
+    label: User Last Modified Record
+    import_template_include: false
+    visible_to_public: false
+    datastore_type: text
+    preview_class: bg-info
+
+  datastore_primary_key: [ref_number, reporting_period]
+  datastore_indexes: ""
+
+  default_preview_sort: ref_number, reporting_period desc, priority_area
+
+  cascading_select_fields: true
+
+  excel_example_height: 50
+
+  triggers:
+  - aistrategy_trigger: |
+      DECLARE
+        errors text[][] := '{{}}';
+        crval RECORD;
+      BEGIN
+        errors := errors || required_error(NEW.ref_number, 'ref_number');
+
+        errors := errors || required_error(NEW.reporting_period, 'reporting_period');
+        errors := errors || choice_error(NEW.reporting_period, {reporting_period}, 'reporting_period');
+
+        errors := errors || required_error(NEW.priority_area, 'priority_area');
+        errors := errors || choice_error(NEW.priority_area, {priority_area}, 'priority_area');
+
+        errors := errors || required_error(NEW.key_action, 'key_action');
+        errors := errors || choice_error(NEW.key_action, {key_action}, 'key_action');
+
+        errors := errors || required_error(NEW.sub_action, 'sub_action');
+        errors := errors || choice_error(NEW.sub_action, {sub_action}, 'sub_action');
+
+        errors := errors || required_error(NEW.activity, 'activity');
```

Comment:
for key_action, sub_action and activity adapt the nap5 sql validation that limits the choices available based on the previous selections, e.g.
```sql
errors := errors || choice_error(NEW.activity,
  array(SELECT unnest from unnest({activity}) where left(unnest, length(NEW.sub_action)) = NEW.sub_action),
  'activity')
```